processing/src/main/java/io/druid/query/search/SearchQueryQueryToolChest.java

```diff
@@ -208,33 +225,82 @@ public Object apply(Result<SearchResultValue> input)
           public Result<SearchResultValue> apply(Object input)
           {
             List<Object> result = (List<Object>) input;
+            boolean needsRename = false;
+            final Map<String, String> outputNameMap = Maps.newHashMap();
+            if (dimensionSpecs != null) {
```

Comment:
The query engine treats empty list and null the same (both mean "include everything as a default dimension") so maybe this should be `if (dimensionSpecs != null && !dimensionSpecs.isEmpty())`